### PR TITLE
Add save() function

### PR DESCRIPTION
**Pull request by diegoscarabelli**
## Description
Add function to build a data node with associated metadata and push it to the registry.

### CODE REVIEW

File: compiler/quilt/tools/util.py [hunk 1]
```diff
@@ -377,3 +378,25 @@ def find_in_package(corenode, path):
...
+         params: dict):
+    """Build and push data to Quilt registry at user/package/data_node,
+    associating params as metadata for the data node.
+    :param user: user name
+    :param package: package name
+    :param data_node: data node name
+    :param data: data to save
+    :param params: metadata
+    """
+    quilt.build(f'{user}/{package}/{data_node}', data)
```
Comment (akarve) [inline compiler/quilt/tools/util.py]: These are not compatible with Python 2.7; and I think that's why CI is blowing up. I _think_ `"{}/{}/{}".format(...)` will work

File: compiler/quilt/__init__.py [hunk 2]
```diff
@@ -119,3 +119,5 @@ def handle_interrupt(signum, stack):
     version_add,
     version_list,
 )
+
+from .tools.util import save
```
Comment (akarve) [inline compiler/quilt/__init__.py]: A blank line at the very end of each file should eliminate the "No EOF" warning we see above

File: compiler/quilt/tools/util.py [hunk 3]
```diff
@@ -377,3 +378,25 @@ def find_in_package(corenode, path):
...
+    pkg = quilt.load(f'{user}/{package}')
+    for key, value in params.items():
+        if isinstance(value, np.ndarray):
+            value = value.astype(float)
+            params[key] = value.tolist()
+    pkg[data_node]._meta.update(params)
+    quilt.build(f'{user}/{package}', pkg)
+    quilt.push(f'{user}/{package}/{data_node}', is_public=True)
```
Comment (akarve) [inline compiler/quilt/tools/util.py]: the `is_public` kwarg should be passed through from the top of the function (i.e. not everyone will want to push the packages `publicly`; similar strategy for other build, push args would be helpful. it is not uncommon to pass `kwargs` through to subroutines, but you may need to filter out inapplicable keywords.

**Comment by akarve** [comment]
A few inline comments. Please bump up to `quilt.util` (instead of `quilt.tools.util`).

**Comment by akarve** [comment]
TODO:
- [ ] investigate numpy imports failing in existing tests

File: compiler/quilt/tools/util.py (hunk 1 above)
Comment (kevinemoore) [reply compiler/quilt/tools/util.py]: Typing also isn't supported in Python2.

**Comment by diegoscarabelli** [comment]
@kevinemoore @akarve I should have implemented the changes that you suggested. It seems that the tests have passed and there are no conflicts. Let me know if there are any further tweaks ti apply.

**Pull request merged**